Classes_Homework_5/fromfile.py

```diff
@@ -0,0 +1,181 @@
+            return source, parsed_text
+        except UnicodeDecodeError:
+            print(cnf.get_values("ERRORS", "cannot_read")+"\n")
+            return None, None
+        except (OSError, ValueError, NoSectionsError, NoValue, PastDate) as err:
+            print(err)
+            return None, None
+
+    def read_source(self, raw_source):
+        """If source file returns no errors, return file path and text, else - ask user to input path to source file"""
+        proper_source, result = self.verify_source(raw_source)  # check path passed by user/default path first
+        inp_request = cnf.get_values("INPUTS", "filepath")
+        while not proper_source:  # if any errors occur with initial source, ask user to input another path as long as it doesn't meet the requirements
+            inp_path = input(inp_request+"\n")
+            proper_source, result = self.verify_source(inp_path)
+        else:
+            return proper_source, result
+
+    def choose_strategy(self):
+        """Choose strategy for normalization to handle possible proper names (like London, Government etc):
+        Option 1: capitalize text ignoring any potential proper names (this will turn London to london, for example)
+        Option 2: don't touch anything that is capitalized in source text, assuming these are proper names
+        """
+        choice1 = cnf.get_values("LABELS", "capital_all")
+        choice2 = cnf.get_values("LABELS", "keep_titles")
+        strategy = pyip.inputMenu(prompt=cnf.get_values("INPUTS", "normalize_strategy")+"\n",
+                                  choices=[choice1, choice2],
+                                  numbered=True
+                                  )
+        return strategy
+
+    def normalize(self, parsed_text, strategy):
+        """Normalize text based on selected strategy"""
+        patt1 = cnf.get_values("PATTERNS", "split_text")
+        if strategy == cnf.get_values("LABELS", "capital_all"):
+            my_text = parsed_text.lower()
+        elif strategy == cnf.get_values("LABELS", "keep_titles"):
+            my_text = s.keep_titles(parsed_text)  # method from Functions strings module
+        return s.capital_first(my_text, patt1)  # method from Functions strings module
+
+    def write(self, formatted_text, source):
+        """Write normalized text to target file. Print info message."""
+        f.append_file(formatted_text, self.target)
+        print(cnf.get_values("MESSAGES", "write_success") % (source, f.get_path(self.target)))
+
+    def remove(self, proper_source):
+        """Remove and print info message"""
+        if proper_source != self.def_source:  # don't delete default source file
+            if type(proper_source) is list:
+                for file in proper_source:
+                    remove(file)
+            else:
+                remove(proper_source)
+            print(cnf.get_values("MESSAGES", "delete_success") % (proper_source))
```

Comment:
A little bit more Pythonic approach instead of printing to  console is writing to log file:

https://docs.python.org/3/howto/logging.html